core/utils.h

```diff
@@ -1946,6 +1946,14 @@ notify(syslog_event_type_t priority, bool internal, bool synch,
 #define SYSLOG_INTERNAL_CRITICAL_ONCE(...) \
     DODEBUG_ONCE(SYSLOG_INTERNAL_CRITICAL(__VA_ARGS__))
 
+#define MSG_COULDNT_SYNCH_WITH_ALL_THREADS_CODE 42
+
+#define fatal_error(id, sub, ...) \
```

Comment:
suggest: add a comment about this not returning -- maybe rename REPORT_FATAL_ERROR_AND_EXIT

Reply:
+1 to REPORT_FATAL_ERROR_AND_EXIT, and with that I put the ASSERT_NOT_REACHED inside of that macro.